dotnet-authserver/src/TeacherIdentity.AuthServer/AuthenticationState.cs

```diff
@@ -617,6 +620,7 @@ public void OnTrnLookupCompleted(string? trn, TrnLookupStatus trnLookupStatus)
 
         Trn = trn;
         TrnLookupStatus = trnLookupStatus;
+        TrnAssociationSource = Models.TrnAssociationSource.Lookup;
```

Comment:
This will set the association source to lookup even while it sets trn to null (if no match is found). Is that ok? Otherwise we can 
`        TrnAssociationSource = trn is null ? null : Models.TrnAssociationSource.Lookup;
`

Reply:
Ideally `TrnAssociationSource` is `null` if `Trn` is `null`.